xla/debug_options_flags.h

```diff
@@ -41,6 +41,12 @@ void AppendDebugOptionsFlags(std::vector<tsl::Flag>* flag_list,
 // Parses the debug option flags from XLA_FLAGS environment variable.
 void ParseDebugOptionFlagsFromEnv();
 
+// Parse the debug options from debug_options file.
```

Comment:
Would be good to describe that this overrides any fields in the global variable returned from `GetDebugOptionsFromFlags` with fields explicitly specified in filename. E.g. something like:

"Given a string containing the textual form of a DebugOptions protobuf, parses it. The global variable containing the debug options returned from 'GetDebugOptionsFromFlags' is mutated by setting fields explicitly specified in the file."

Somewhat unrelatedly, probably worth moving GetNonDefaultDebugOptions to this file. It doesn't actually dump antyhing to disk, and is closely tied with this function. But that can be done in a separate PR